### PR TITLE
[build] Add protobuf compiler plugin and required dependencies for generating gRPC Java stubs

### PR DESCRIPTION
**Pull request by rkunds**
## How was this PR tested?

wpi: Local and GHCI

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: services/venice-server/build.gradle [hunk 1]
```diff
@@ -63,6 +70,25 @@ jar {
   }
 }
 
+def grpcVersion = '1.54.1'
+def protobufVersion = '3.21.7'
+
+protobuf {
```
Comment (sushantmane) [inline services/venice-server/build.gradle:76]: let's move this to the parent `build.gradle` and then make this plugin and task available via `subprojects {..}`

File: build.gradle [hunk 2]
```diff
@@ -114,7 +119,13 @@ ext.libraries = [
...
     zkclient: 'com.101tec:zkclient:0.7', // For Kafka AdminUtils
     zookeeper: 'org.apache.zookeeper:zookeeper:3.5.9',
-    zstd: 'com.github.luben:zstd-jni:1.5.2-3'
+    zstd: 'com.github.luben:zstd-jni:1.5.2-3',
+
+    grpcNettyShaded: 'io.grpc:grpc-netty-shaded:' + grpcVersion,
+    grpcProtobuf: 'io.grpc:grpc-protobuf:' + grpcVersion,
+    grpcStub: 'io.grpc:grpc-stub:' + grpcVersion,
+    tomcatAnnotations: 'org.apache.tomcat:annotations-api:6.0.53',
+    guava: 'com.google.guava:guava:30.1-jre'
```
Comment (sushantmane) [inline build.gradle:128]: We don't need `guava` dependency for gRPC. Let's remove it.

File: build.gradle [hunk 3]
```diff
@@ -201,7 +212,6 @@ subprojects {
     }
     compileOnly {
       // These dependencies are transitively used at runtime, so we cannot exclude them further than compileOnly
-      exclude group: 'com.google.guava'
```
Comment (sushantmane) [inline build.gradle:204]: this might cause dependency conflict. Let's not remove this line

**Review by sushantmane — CHANGES_REQUESTED** [review]
Congrats on your first PR! I've left some comments.

File: build.gradle [hunk 4]
```diff
@@ -114,7 +119,13 @@ ext.libraries = [
     xerces: 'xerces:xercesImpl:2.9.1',
     zkclient: 'com.101tec:zkclient:0.7', // For Kafka AdminUtils
     zookeeper: 'org.apache.zookeeper:zookeeper:3.5.9',
-    zstd: 'com.github.luben:zstd-jni:1.5.2-3'
+    zstd: 'com.github.luben:zstd-jni:1.5.2-3',
+
+    grpcNettyShaded: 'io.grpc:grpc-netty-shaded:' + grpcVersion,
+    grpcProtobuf: 'io.grpc:grpc-protobuf:' + grpcVersion,
+    grpcStub: 'io.grpc:grpc-stub:' + grpcVersion,
+    tomcatAnnotations: 'org.apache.tomcat:annotations-api:6.0.53',
```
Comment (adamxchen) [inline build.gradle:127]: is this dependency needed? Seems nothing related to gPRC. 😕

File: services/venice-server/build.gradle (hunk 1 above)
Comment (rkunds) [reply services/venice-server/build.gradle]: done, moved dependencies and tasks to parent `build.gradle` file.

File: build.gradle (hunk 4 above)
Comment (rkunds) [reply build.gradle:125]: To my understanding it should be necessary to include this as a `compileOnly` dependency to obtain annotations for generated code. For reference https://github.com/grpc/grpc-java/issues/9179.

Comment (adamxchen) [reply build.gradle:125]: Got it.

**Review by sushantmane — APPROVED** [review]
LGTM. Thanks, @rkunds!

**Pull request merged**